scripts/firedrake-install

```diff
@@ -1347,8 +1334,9 @@ def build_and_install_torch():
         raise InstallError("CUDA installation is not available on MacOS.")
     if sys.version_info >= (3, 10):
         run_pip_install(["typing_extensions"])
-    extra_index_url = ["--extra-index-url", "https://download.pytorch.org/whl/cpu"] if args.torch == "cpu" else []
-    run_pip_install(["torch"] + extra_index_url)
+    # FIXME: use nightly to see if that makes numpy 2.0 happy
```

Comment:
Should really be
```
pip install torch==2.3.0 --index-url https://download.pytorch.org/whl/test/cpu
```
as detailed in https://dev-discuss.pytorch.org/t/pytorch-release-2-3-0-final-rc-is-available/1995

2.3.0 is supposed to be released 24/04.